# Euler Project Challenges

In this document I will be contributing solutions to challenges in the Euler project. I will mostly work though in order and will include links to the questions or copy out the question text.



In [3]:
##hello 
import numpy as np



## Problem 1
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.

In [5]:
limit =100
a = 3
b = 5
factors = 0
factors_array = []

for i in range(limit):
    div_a = i%a
    div_b = i%b
    if div_a == 0:
        factors += i
        factors_array.append(i)
        continue
    elif div_b == 0:
        factors += i
        factors_array.append(i)
        continue

print(factors)
print(factors_array)

2318
[0, 3, 5, 6, 9, 10, 12, 15, 18, 20, 21, 24, 25, 27, 30, 33, 35, 36, 39, 40, 42, 45, 48, 50, 51, 54, 55, 57, 60, 63, 65, 66, 69, 70, 72, 75, 78, 80, 81, 84, 85, 87, 90, 93, 95, 96, 99]


### Answer  -  2318

Answer = 233168


### Problem 2
Each new term in the Fibonacci sequence is generated by adding the previous two terms. By starting with 1 and 2, the first 10 terms will be:

1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...

By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

In [10]:
limit = 4000000
fib_seq = [1,2]
fib_sum = 0

## Generating the fibonacci sequence up to the defined limit
for i in range(limit):
    fib_var = fib_seq[i] + fib_seq[i+1]
    if fib_var <= limit:
        fib_seq.append(fib_var)
    else: break
print(fib_seq)

## summing the even values in this range
for i in fib_seq:
    even_test = i%2
    if even_test == 0:
        fib_sum += i
print(fib_sum)

[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946, 17711, 28657, 46368, 75025, 121393, 196418, 317811, 514229, 832040, 1346269, 2178309, 3524578]
4613732


#### Answer : 4613732